### PR TITLE
docs: Fix typos in Discovery.md

### PR DESCRIPTION
**Pull request by steverfrancis**
Fix typos.

Signed-off-by: Steve Francis <steve.francis@talos-systems.com>

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5390)

### CODE REVIEW

File: website/content/v1.1/learn-more/discovery.md
Comment (frezbo) [inline website/content/v1.1/learn-more/discovery.md:7]: this is little confusing, there is no `shared and unique` :sweat_smile: I think a shared service makes sense

Comment (steverfrancis) [reply website/content/v1.1/learn-more/discovery.md:7]: I assume it's shared within the cluster, but globally unique, yes? If so, I can clarify

Comment (frezbo) [reply website/content/v1.1/learn-more/discovery.md:7]: I think yes, but @Ulexus can clarify

Comment (smira) [reply website/content/v1.1/learn-more/discovery.md:7]: yes, it's same key for all cluster nodes, but globally unique, so that other clusters have different keys (and can't see each other information)

Comment (steverfrancis) [reply website/content/v1.1/learn-more/discovery.md]: Clarified

**Review by frezbo — APPROVED** [review]

**Review by smira — APPROVED** [review]

**Comment by smira** [comment]
let's please wait until #5401 is merged

**Review by Unix4ever — APPROVED** [review]

**Pull request merged**
